[Bug]: Crash on 5.0.0 - "Unrecognized operation: login-user-from-subscription"

### What happened?

I see crash on few devices (Android 11 and 10 for sure).
This is on calling `OneSignal.initWithContext()`



### Steps to reproduce?

```Markdown
-
```


### What did you expect to happen?

-

### OneSignal Android SDK version

5.0.0

### Android version

11, 10

### Specific Android models

```Markdown
* Galaxy A31
* Galaxy A7 (2018)
* Galaxy S9+
* Moto G(9) Plus
* Nokia 9
```


### Relevant log output

```Shell
Stacktrace:

Caused by java.lang.reflect.InvocationTargetException:
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:90)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:83)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceProvider.getService$core_release(ServiceProvider.kt:40)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:197)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:126)
       at pl.tajchert.canary.CanaryApp.onCreate(CanaryApp.kt:66)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7593)
       at android.app.ActivityThread.access$1500(ActivityThread.java:301)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2177)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8653)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)

Caused by java.lang.Exception: Unrecognized operation: login-user-from-subscription
       at com.onesignal.core.internal.operations.impl.OperationModelStore.create(OperationModelStore.kt:58)
       at com.onesignal.core.internal.operations.impl.OperationModelStore.create(OperationModelStore.kt:27)
       at com.onesignal.common.modeling.ModelStore.load(ModelStore.kt:129)
       at com.onesignal.core.internal.operations.impl.OperationModelStore.<init>(OperationModelStore.kt:30)
       at java.lang.reflect.Constructor.newInstance0(Constructor.java)
       at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:90)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:83)
       at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
       at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
       at com.onesignal.common.services.ServiceProvider.getService$core_release(ServiceProvider.kt:40)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:197)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:126)
       at pl.tajchert.canary.CanaryApp.onCreate(CanaryApp.kt:66)
       at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:7593)
       at android.app.ActivityThread.access$1500(ActivityThread.java:301)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2177)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:246)
       at android.app.ActivityThread.main(ActivityThread.java:8653)
       at java.lang.reflect.Method.invoke(Method.java)
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

<!-- probot = {"onesignal-probot":{"response_time_in_business_days":1}} -->

I am having same issue as well. Checked on Android 13.
Here is logcat:
````
08-27 17:41:40.205  9446  9446 W System.err:  java.lang.reflect.InvocationTargetException
08-27 17:41:40.205  9446  9446 W System.err: 	at java.lang.reflect.Constructor.newInstance0(Native Method)
08-27 17:41:40.205  9446  9446 W System.err: 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(Unknown Source:217)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(Unknown Source:50)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.common.services.ServiceProvider.getService(Unknown Source:5)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(Unknown Source:187)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(Unknown Source:50)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.common.services.ServiceProvider.getService(Unknown Source:5)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.internal.OneSignalImp.initWithContext(Unknown Source:78)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.onesignal.OneSignal.initWithContext(Unknown Source:16)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.sunny.push.OneSignalPush.Init(Unknown Source:11)
08-27 17:41:40.205  9446  9446 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.expr.PrimProcedure.apply(Unknown Source:182)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.mapping.CallContext.runUntilDone(Unknown Source:8)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.mapping.CallContext.runUntilValue(Unknown Source:10)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.mapping.MethodProc.applyN(Unknown Source:11)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.kawa.reflect.Invoke.applyN(Unknown Source:237)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.kawa.functions.ApplyToArgs.applyN(Unknown Source:17)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.kawa.functions.Apply.applyN(Unknown Source:7)
08-27 17:41:40.205  9446  9446 W System.err: 	at gnu.mapping.ProcedureN.apply2(Unknown Source:9)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.google.youngandroid.runtime.callComponentMethod(Unknown Source:57)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.appzard.vknow360.Push.Screen1.Button1$Click(Unknown Source:15)
08-27 17:41:40.205  9446  9446 W System.err: 	at com.appzard.vknow360.Push.Screen1$frame.apply0(Unknown Source:10)
08-27 17:41:40.206  9446  9446 W System.err: 	at gnu.expr.ModuleBody.applyN(Unknown Source:72)
08-27 17:41:40.206  9446  9446 W System.err: 	at gnu.expr.ModuleMethod.applyN(Unknown Source:2)
08-27 17:41:40.206  9446  9446 W System.err: 	at gnu.kawa.functions.ApplyToArgs.applyN(Unknown Source:17)
08-27 17:41:40.206  9446  9446 W System.err: 	at gnu.kawa.functions.Apply.applyN(Unknown Source:7)
08-27 17:41:40.206  9446  9446 W System.err: 	at gnu.mapping.ProcedureN.apply2(Unknown Source:9)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.appzard.vknow360.Push.Screen1.dispatchEvent(Unknown Source:28)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.google.appinventor.components.runtime.EventDispatcher.dispatchEvent(Unknown Source:10)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.google.appinventor.components.runtime.Button.Click(Unknown Source:5)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.google.appinventor.components.runtime.Button.click(Unknown Source:0)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.google.appinventor.components.runtime.ButtonBase.onClick(Unknown Source:0)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.view.View.performClick(View.java:7590)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.view.View.performClickInternal(View.java:7567)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.view.View$PerformClick.run(View.java:29724)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.os.Handler.handleCallback(Handler.java:942)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.os.Handler.dispatchMessage(Handler.java:99)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.os.Looper.loopOnce(Looper.java:223)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.os.Looper.loop(Looper.java:324)
08-27 17:41:40.206  9446  9446 W System.err: 	at android.app.ActivityThread.main(ActivityThread.java:8378)
08-27 17:41:40.206  9446  9446 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:582)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1059)
08-27 17:41:40.206  9446  9446 W System.err: Caused by: java.lang.ExceptionInInitializerError
08-27 17:41:40.206  9446  9446 W System.err: 	at kotlinx.coroutines.channels.ChannelKt.Channel(Unknown Source:63)
08-27 17:41:40.206  9446  9446 W System.err: 	at kotlinx.coroutines.channels.ChannelKt.Channel$default(Unknown Source:16)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.onesignal.common.threading.Waiter.<init>(Unknown Source:6)
08-27 17:41:40.206  9446  9446 W System.err: 	at com.onesignal.core.internal.preferences.impl.PreferencesService.<init>(Unknown Source:4)
08-27 17:41:40.206  9446  9446 W System.err: 	... 45 more
08-27 17:41:40.206  9446  9446 W System.err: Caused by: java.lang.ClassCastException
08-27 17:41:40.206  9446  9446 W System.err: 	at java.util.concurrent.atomic.AtomicReferenceFieldUpdater$AtomicReferenceFieldUpdaterImpl.<init>(AtomicReferenceFieldUpdater.java:365)
08-27 17:41:40.206  9446  9446 W System.err: 	at java.util.concurrent.atomic.AtomicReferenceFieldUpdater.newUpdater(AtomicReferenceFieldUpdater.java:116)
08-27 17:41:40.206  9446  9446 W System.err: 	at kotlinx.coroutines.channels.AbstractSendChannel.<clinit>(Unknown Source:6)
08-27 17:41:40.206  9446  9446 W System.err: 	... 49 more

````

--- a/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/modeling/IModelStore.kt
+++ b/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/modeling/IModelStore.kt
@@ -16,7 +16,7 @@ interface IModelStore<TModel> :
      *
      * @param jsonObject The optional [JSONObject] to initialize the new model with.
      */
-    fun create(jsonObject: JSONObject? = null): TModel
+    fun create(jsonObject: JSONObject? = null): TModel?
 
     /**
      * List the models that are owned by this model store.

--- a/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/modeling/ModelStore.kt
+++ b/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/modeling/ModelStore.kt
@@ -126,7 +126,7 @@ abstract class ModelStore<TModel>(
             val str = _prefs.getString(PreferenceStores.ONESIGNAL, PreferenceOneSignalKeys.MODEL_STORE_PREFIX + name, "[]")
             val jsonArray = JSONArray(str)
             for (index in 0 until jsonArray.length()) {
-                val newModel = create(jsonArray.getJSONObject(index))
+                val newModel = create(jsonArray.getJSONObject(index)) ?: continue
                 _models.add(newModel)
                 // listen for changes to this model
                 newModel.subscribe(this)

--- a/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/modeling/SingletonModelStore.kt
+++ b/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/modeling/SingletonModelStore.kt
@@ -24,7 +24,7 @@ open class SingletonModelStore<TModel>(
                 return model
             }
 
-            val createdModel = store.create()
+            val createdModel = store.create() ?: throw Exception("Unable to initialize model from store $store")
             createdModel.id = _singletonId
             store.add(createdModel)
             return createdModel

--- a/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/operations/impl/OperationModelStore.kt
+++ b/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/operations/impl/OperationModelStore.kt
@@ -3,11 +3,13 @@ package com.onesignal.core.internal.operations.impl
 import com.onesignal.common.modeling.ModelStore
 import com.onesignal.core.internal.operations.Operation
 import com.onesignal.core.internal.preferences.IPreferencesService
+import com.onesignal.debug.internal.logging.Logging
 import com.onesignal.user.internal.operations.CreateSubscriptionOperation
 import com.onesignal.user.internal.operations.DeleteAliasOperation
 import com.onesignal.user.internal.operations.DeleteSubscriptionOperation
 import com.onesignal.user.internal.operations.DeleteTagOperation
 import com.onesignal.user.internal.operations.LoginUserOperation
+import com.onesignal.user.internal.operations.LoginUserFromSubscriptionOperation
 import com.onesignal.user.internal.operations.RefreshUserOperation
 import com.onesignal.user.internal.operations.SetAliasOperation
 import com.onesignal.user.internal.operations.SetPropertyOperation
@@ -18,6 +20,7 @@ import com.onesignal.user.internal.operations.TrackSessionStartOperation
 import com.onesignal.user.internal.operations.TransferSubscriptionOperation
 import com.onesignal.user.internal.operations.UpdateSubscriptionOperation
 import com.onesignal.user.internal.operations.impl.executors.IdentityOperationExecutor
+import com.onesignal.user.internal.operations.impl.executors.LoginUserFromSubscriptionOperationExecutor
 import com.onesignal.user.internal.operations.impl.executors.LoginUserOperationExecutor
 import com.onesignal.user.internal.operations.impl.executors.RefreshUserOperationExecutor
 import com.onesignal.user.internal.operations.impl.executors.SubscriptionOperationExecutor
@@ -30,13 +33,15 @@ internal class OperationModelStore(prefs: IPreferencesService) : ModelStore<Oper
         load()
     }
 
-    override fun create(jsonObject: JSONObject?): Operation {
+    override fun create(jsonObject: JSONObject?): Operation? {
         if (jsonObject == null) {
-            throw NullPointerException("jsonObject")
+            Logging.error("null jsonObject sent to OperationModelStore.create")
+            return null
         }
 
         if (!jsonObject.has(Operation::name.name)) {
-            throw IllegalArgumentException("jsonObject must have '${Operation::name.name}' attribute")
+            Logging.error("jsonObject must have '${Operation::name.name}' attribute")
+            return null
         }
 
         // Determine the type of operation based on the name property in the json
@@ -48,6 +53,7 @@ internal class OperationModelStore(prefs: IPreferencesService) : ModelStore<Oper
             SubscriptionOperationExecutor.DELETE_SUBSCRIPTION -> DeleteSubscriptionOperation()
             SubscriptionOperationExecutor.TRANSFER_SUBSCRIPTION -> TransferSubscriptionOperation()
             LoginUserOperationExecutor.LOGIN_USER -> LoginUserOperation()
+            LoginUserFromSubscriptionOperationExecutor.LOGIN_USER_FROM_SUBSCRIPTION_USER -> LoginUserFromSubscriptionOperation()
             RefreshUserOperationExecutor.REFRESH_USER -> RefreshUserOperation()
             UpdateUserOperationExecutor.SET_TAG -> SetTagOperation()
             UpdateUserOperationExecutor.DELETE_TAG -> DeleteTagOperation()

